core/trino-main/src/main/java/io/trino/sql/planner/LocalExecutionPlanner.java

```diff
@@ -594,31 +562,65 @@ private LocalExecutionPlanContext(
 
         public void addDriverFactory(boolean inputDriver, boolean outputDriver, PhysicalOperation physicalOperation, OptionalInt driverInstances)
         {
-            addDriverFactory(
-                    inputDriver,
-                    outputDriver,
-                    physicalOperation.getOperatorFactories(),
-                    driverInstances,
-                    physicalOperation.getPipelineExecutionStrategy());
+            List<OperatorFactory> operatorFactories = physicalOperation.getOperatorFactories();
+            validateFirstOperatorFactory(inputDriver, operatorFactories.get(0), physicalOperation.getPipelineExecutionStrategy());
+            addLookupOuterDrivers(outputDriver, operatorFactories);
+            operatorFactories = handleLateMaterialization(operatorFactories);
```

Comment:
I think this will be easier to read if it is inlined with a comment.  This makes it clear here that we are only modifying the  operator factories if late materialization is enabled.
```java
{
    List<OperatorFactory> operatorFactories = physicalOperation.getOperatorFactories();
    validateFirstOperatorFactory(inputDriver, operatorFactories.get(0), physicalOperation.getPipelineExecutionStrategy());
    addLookupOuterDrivers(outputDriver, operatorFactories);
   
    // handle late materialization
    if (isLateMaterializationEnabled(taskContext.getSession())) {
        operatorFactories = WorkProcessorPipelineSourceOperator.convertOperators(operatorFactories);
    }
    driverFactories.add(new DriverFactory(
            getNextPipelineId(), 
            inputDriver, 
            outputDriver, 
            operatorFactories,
            driverInstances, 
            physicalOperation.getPipelineExecutionStrategy()));
}
````